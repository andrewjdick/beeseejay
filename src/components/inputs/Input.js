import React from "react";
import { string } from "prop-types";
import { StyledInput } from "./styles";

export const Input = ({ type, ...props }) => (
  <StyledInput type={type} {...props} />
);

Input.defaultProps = {
  type: "text"
};

Input.proptypes = {
  type: string
};
