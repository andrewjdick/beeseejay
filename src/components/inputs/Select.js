import React from "react";
import { arrayOf, shape, string } from "prop-types";

export const Select = ({ options, ...props }) => (
  <select {...props}>
    {options.map(({ value, label }) => (
      <option key={value} value={value}>
        {label}
      </option>
    ))}
  </select>
);

Select.propTypes = {
  options: arrayOf(
    shape({
      value: string,
      label: string
    })
  ).isRequired()
};
