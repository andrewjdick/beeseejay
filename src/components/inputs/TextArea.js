import React from "react";
import { string } from "prop-types";
import { StyledTextArea } from "./styles";

export const MAX_CHARACTER_COUNT = "150";

export const TextArea = ({ maxLength, ...props }) => (
  <StyledTextArea maxLength={maxLength} {...props} />
);

TextArea.defaultProps = {
  maxLength: MAX_CHARACTER_COUNT
};

TextArea.propTypes = {
  maxLength: string
};
