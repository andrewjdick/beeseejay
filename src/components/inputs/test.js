import React from "react";
import renderer from "react-test-renderer";
import { Input } from "./Input";
import { Select } from "./Select";
import { TextArea } from "./TextArea";

it("renders input correctly", () => {
  const input = renderer.create(<Input />).toJSON();
  expect(input).toMatchSnapshot();
});

it("renders select correctly", () => {
  const select = renderer
    .create(<Select options={[{ key: "123", value: "456" }]} />)
    .toJSON();
  expect(select).toMatchSnapshot();
});

it("renders textArea correctly", () => {
  const textArea = renderer.create(<TextArea />).toJSON();
  expect(textArea).toMatchSnapshot();
});
