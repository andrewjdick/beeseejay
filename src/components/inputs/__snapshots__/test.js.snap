// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<input
  className="sc-bdVaJa bVFsUj"
  type="text"
/>
`;

exports[`Select renders correctly 1`] = `
<select>
  <option
    value="456"
  />
  <option
    value="def"
  />
</select>
`;

exports[`TextArea renders correctly 1`] = `
<textarea
  className="sc-bwzfXH eBguEi"
  maxLength="150"
/>
`;
